# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditUtils.cpp
# Commit: a4ef62ad447f
# Full Hash: a4ef62ad447f3bdeca250ebc7052854226c8891f
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231588
# ==============================================================================

diff -r fb611166c7ab -r a4ef62ad447f editor/libeditor/HTMLEditUtils.cpp
--- a/editor/libeditor/HTMLEditUtils.cpp	Thu May 08 13:37:20 2025 +0000
+++ b/editor/libeditor/HTMLEditUtils.cpp	Thu May 08 13:37:20 2025 +0000
@@ -2829,9 +2829,10 @@
   // Use range boundaries and RangeUtils::CompareNodeToRange() to compare
   // selection start to new block.
   bool nodeBefore, nodeAfter;
-  nsresult rv = RangeUtils::CompareNodeToRangeBoundaries(
-      const_cast<Element*>(&aElement), aCurrentPoint.ToRawRangeBoundary(),
-      aCurrentPoint.ToRawRangeBoundary(), &nodeBefore, &nodeAfter);
+  nsresult rv =
+      RangeUtils::CompareNodeToRangeBoundaries<TreeKind::ShadowIncludingDOM>(
+          const_cast<Element*>(&aElement), aCurrentPoint.ToRawRangeBoundary(),
+          aCurrentPoint.ToRawRangeBoundary(), &nodeBefore, &nodeAfter);
   if (NS_FAILED(rv)) {
     NS_WARNING("RangeUtils::CompareNodeToRange() failed");
     return Err(rv);